service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java

```diff
@@ -135,8 +156,12 @@ public OrderTypeAndDocument buildOrderTypeAndDocument(OrderTypeAndDocument typeA
         return orderTemplateBuilder.build();
     }
 
-    public String generateOrderDocumentFileName(String type) {
-        return type.toLowerCase().replaceAll("[()]", "").replaceAll("[ ]", "_") + ".pdf";
+    public String generateOrderDocumentFileName(GeneratedOrderType orderType, GeneratedOrderSubtype orderSubtype) {
+        String type = orderType.getLabel().toLowerCase().replaceAll("[()]", "").replaceAll("[ ]",
```

Comment:
We could chain `replace` calls here, because `replaceAll` creates `Pattern` and `Matcher` instances on every call.

Reply:
so do you mean:
'       
String type = orderType.getLabel().toLowerCase().replace("(", "").replace(")", "").replace(" ", "_");
'

Reply:
turns our that replace does replace all, so this works 👍